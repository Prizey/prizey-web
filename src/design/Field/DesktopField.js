import React from 'react'
import { InputLabel, Input, FormHelperText } from '@material-ui/core'

export default ({ field, form, type = 'text', label, ...props }) => (
  <React.Fragment>
    <InputLabel htmlFor={field.name}>{label}</InputLabel>
    <Input {...props} {...field} type={type} id={field.name} />
    <FormHelperText id={`${field.name}-text`}>
      {form.touched[field.name] && form.errors[field.name]}
    </FormHelperText>
  </React.Fragment>
)
