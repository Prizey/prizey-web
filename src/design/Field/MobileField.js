import React from 'react'
import { Input, FormHelperText } from '@material-ui/core'

export default ({ field, form, type = 'text', ...props }) => (
  <React.Fragment>
    <Input
      {...props}
      {...field}
      type={type}
      id={field.name}
      placeholder={
        props.placeholder ? `${props.label} ${props.placeholder}` : props.label
      }
    />
    <FormHelperText id={`${field.name}-text`}>
      {form.touched[field.name] && form.errors[field.name]}
    </FormHelperText>
  </React.Fragment>
)
