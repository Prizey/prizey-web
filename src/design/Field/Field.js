import React from 'react'
import withWidth, { isWidthUp } from '@material-ui/core/withWidth'
import { FormControl } from '@material-ui/core'

import MobileField from './MobileField'
import DesktopField from './DesktopField'

export const InputField = ({ width, field, form, ...props }) => {
  const UseField = isWidthUp('sm', width) ? DesktopField : MobileField
  return (
    <FormControl
      component="fieldset"
      error={form.touched[field.name] && !!form.errors[field.name]}
      id={`${field.name}-fieldset`}
      fullWidth
    >
      <UseField {...props} field={field} form={form} />
    </FormControl>
  )
}

export default withWidth()(InputField)
