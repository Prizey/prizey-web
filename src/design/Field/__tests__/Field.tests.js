import React from 'react'
import renderer from 'react-test-renderer'

import { InputField as Field } from '../Field'

it('renders correctly the desktop field', () => {
  const tree = renderer
    .create(
      <Field
        label="foo"
        form={{ errors: {}, touched: {} }}
        field={{ name: 'foo' }}
        width="md"
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly the desktop field with error', () => {
  const tree = renderer
    .create(
      <Field
        label="foo"
        form={{ errors: { foo: 'bar' }, touched: { foo: true } }}
        field={{ name: 'foo' }}
        width="md"
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly the mobile field', () => {
  const tree = renderer
    .create(
      <Field
        label="foo"
        form={{ errors: {}, touched: {} }}
        field={{ name: 'foo' }}
        width="xs"
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly the mobile field with error', () => {
  const tree = renderer
    .create(
      <Field
        label="foo"
        form={{ errors: { foo: 'bar' }, touched: { foo: true } }}
        field={{ name: 'foo' }}
        width="xs"
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
