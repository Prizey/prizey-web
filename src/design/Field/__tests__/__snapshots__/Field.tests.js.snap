// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the desktop field 1`] = `
<fieldset
  className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
  id="foo-fieldset"
>
  <label
    className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
    data-shrink={false}
    htmlFor="foo"
  >
    foo
  </label>
  <div
    className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-formControl-37 MuiInput-formControl-24"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input-46 MuiInput-input-31"
      disabled={false}
      id="foo"
      name="foo"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
  <p
    className="MuiFormHelperText-root-53"
    id="foo-text"
  />
</fieldset>
`;

exports[`renders correctly the desktop field with error 1`] = `
<fieldset
  className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
  id="foo-fieldset"
>
  <label
    className="MuiFormLabel-root-16 MuiFormLabel-error-19 MuiInputLabel-error-8 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
    data-shrink={false}
    htmlFor="foo"
  >
    foo
  </label>
  <div
    className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-error-42 MuiInput-error-28 MuiInputBase-formControl-37 MuiInput-formControl-24"
    onClick={[Function]}
  >
    <input
      aria-invalid={true}
      className="MuiInputBase-input-46 MuiInput-input-31"
      disabled={false}
      id="foo"
      name="foo"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
  <p
    className="MuiFormHelperText-root-53 MuiFormHelperText-error-54"
    id="foo-text"
  >
    bar
  </p>
</fieldset>
`;

exports[`renders correctly the mobile field 1`] = `
<fieldset
  className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
  id="foo-fieldset"
>
  <div
    className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-formControl-37 MuiInput-formControl-24"
    label="foo"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input-46 MuiInput-input-31"
      disabled={false}
      id="foo"
      name="foo"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="foo"
      required={false}
      type="text"
    />
  </div>
  <p
    className="MuiFormHelperText-root-53"
    id="foo-text"
  />
</fieldset>
`;

exports[`renders correctly the mobile field with error 1`] = `
<fieldset
  className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
  id="foo-fieldset"
>
  <div
    className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-error-42 MuiInput-error-28 MuiInputBase-formControl-37 MuiInput-formControl-24"
    label="foo"
    onClick={[Function]}
  >
    <input
      aria-invalid={true}
      className="MuiInputBase-input-46 MuiInput-input-31"
      disabled={false}
      id="foo"
      name="foo"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="foo"
      required={false}
      type="text"
    />
  </div>
  <p
    className="MuiFormHelperText-root-53 MuiFormHelperText-error-54"
    id="foo-text"
  >
    bar
  </p>
</fieldset>
`;
